pkg/draft/pack/load.go

```diff
@@ -1,49 +1,201 @@
 package pack
 
 import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"errors"
 	"fmt"
+	"io"
 	"io/ioutil"
+	"os"
+	"path"
 	"path/filepath"
+	"strings"
 
+	"github.com/facebookgo/symwalk"
+	"github.com/ghodss/yaml"
 	"k8s.io/helm/pkg/chartutil"
-
-	"github.com/Azure/draft/pkg/osutil"
 )
 
+// BufferedFile represents an archive file buffered for later processing.
+type BufferedFile struct {
+	Name string
+	Data []byte
+}
+
 // FromDir takes a string name, tries to resolve it to a file or directory, and then loads it.
 //
 // This is the preferred way to load a pack. It will discover the pack encoding
 // and hand off to the appropriate pack reader.
 func FromDir(dir string) (*Pack, error) {
-	pack := new(Pack)
 
 	topdir, err := filepath.Abs(dir)
 	if err != nil {
 		return nil, err
 	}
+	prefix := topdir + string(filepath.Separator)
+
+	files := []*BufferedFile{}
```

Comment:
Instead of bringing in our own re-implementation of a BufferedFile, would it be OK if we use a [`chartutil.BufferedFile`](https://github.com/kubernetes/helm/blob/d4e129d55c43bb66f9da03ab0abd1482725eec0d/pkg/chartutil/load.go#L59-L63) instead? It seems like we're sharing a lot of their implementation here which is completely fine (and encouraged!), but I'd like to see if we can reduce code duplication where it makes sense. :)